components/movementsensor/gpsutils/cachedData_test.go

```diff
@@ -60,3 +63,171 @@ func TestReadingsSerial(t *testing.T) {
 	test.That(t, err, test.ShouldBeNil)
 	test.That(t, fix1, test.ShouldEqual, fix)
 }
+
+func TestPosition(t *testing.T) {
+	ctx := context.Background()
+	logger := logging.NewTestLogger(t)
+	g := NewCachedData(&mockDataReader{}, logger)
+
+	t.Run("no current location", func(t *testing.T) {
+		g.lastPosition.SetLastPosition(geo.NewPoint(32.4, 54.2))
+		g.nmeaData = NmeaParser{
+			Location: nil,
+		}
+
+		expectedPoint := geo.NewPoint(32.4, 54.2)
+
+		pos, alt, err := g.Position(ctx, make(map[string]interface{}))
+		test.That(t, err, test.ShouldNotBeNil)
```

Comment:
test the exact error value here instead of ShouldNotBeNil

Reply:
@susmitaSanyal Yeah, I noticed this too, but the error is just the string "nil gps location, check nmea message parsing", not sure if that will ever change. I will fix this regardless.